src/connection.c

```diff
@@ -579,6 +580,10 @@ int wait_connection_ready (create_connection_ctx_t *ctx)
 	OnboardLog("Received Unauthorized response with status: %d\n", wait_status);
     return WAIT_ACTION_RETRY;
   }
+  if (!server_is_null (&ctx->server_list->redirect)) {
+    ParodusError("Client connection timeout after redirect\n");	
```

Comment:
We need check for redirect response not received, so server_list->redirect will be NULL for the issue case

Reply:
When we reach this line, we know that it's not a redirect response (302, 303 or 307) because that condition is captured in the above paragraph, and returns, one way or the other.

Reply:
I feel better to confirm that wait_status is 0 and server_list->redirect is NULL
Please recheck the ! in the condition.
Aslo reword this "Client connection timeout after redirect" as it indicates like redirect was received.